### PR TITLE
fix: add permission 'jenkins/labelsData' to pipline manage role

### PR DESCRIPTION
**Pull request by 123liubao**
Fix: https://github.com/kubesphere/ks-devops/issues/430

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
> /cc @LinuxSuRen @JohnNiang

Thanks for ping us. But a better way is to ping the DevOps SIG team.

/cc @kubesphere/sig-devops

File: roles/ks-core/prepare/files/ks-init/role-templates.yaml [hunk 1]
```diff
@@ -3169,6 +3169,7 @@ role:
         - 'pipelines/consolelog'
         - 'pipelines/scan'
         - 'pipelines/sonarstatus'
+        - 'jenkins/labelsData'
```
Comment (LinuxSuRen) [inline roles/ks-core/prepare/files/ks-init/role-templates.yaml:3172]: This role already contains this resource permission.

```
iam.kubesphere.io/dependencies: '["role-template-view-pipelines", "role-template-manage-pipelineruns", "role-template-view-credentials"]'
```

See also https://github.com/kubesphere/ks-installer/blob/f860029d1d974698234d6aa249425e269af8e491/roles/ks-core/prepare/files/ks-init/role-templates.yaml#L3239

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Thanks for your contribution.

In https://github.com/kubesphere/ks-devops/issues/430 , he met the permission issue with `artifact download` instead of Jenkins agent lable.

File: roles/ks-core/prepare/files/ks-init/role-templates.yaml (hunk 1 above)
Comment (123liubao) [reply roles/ks-core/prepare/files/ks-init/role-templates.yaml:3172]: OK, I would reproduce that issue first.

**Comment by 123liubao** [comment]
@LinuxSuRen 
I have reproduced this issue, artifact download use the API 
`/kapis/devops.kubesphere.io/v1alpha2/jenkins/job/cly-devops7846x/job/test-pipeline/2/artifact/target/devops-sample-0.0.1-SNAPSHOT.jar` to download and meet the following error:
image

according to the error info , user cannot get resource 'jenkins/cly-devops7846x'  , but `/job/cly-devops7846x/job/test-pipeline/2/artifact/target/devops-sample-0.0.1-SNAPSHOT.jar` is jenkins url , so maybe we can not only add permissions to roles, but also update the rules of the url parse.
image

**Comment by LinuxSuRen** [comment]
hi @123liubao , thanks for your deep research on this issue.

I have a new idea. IMO, the following API is pretty ugly. The most important reason is that it is kind of a proxy of the Jenkins API. It's not necessary to expose any Jenkins-related path here.

`/kapis/devops.kubesphere.io/v1alpha2/jenkins/job/cly-devops7846x/job/test-pipeline/2/artifact/target/devops-sample-0.0.1-SNAPSHOT.jar`

So, how about creating a new API for this. For instance:

`GET /kapis/devops.kubesphere.io/v1alpha2/pipelineruns/demo-artifact-tt85n/artifacts?filePath=xxx.log`

* pros
  * Easy to understand, easy to memory, shorter
  * Less to expose the backend tech
  * Have the same permission control pattern (`pipelineruns/artifacts`)
* cons
  * Extra work: both the backend and frontend

**Comment by 123liubao** [comment]
> hi @123liubao , thanks for your deep research on this issue.
> 
> I have a new idea. IMO, the following API is pretty ugly. The most important reason is that it is kind of a proxy of the Jenkins API. It's not necessary to expose any Jenkins-related path here.
> 
> `/kapis/devops.kubesphere.io/v1alpha2/jenkins/job/cly-devops7846x/job/test-pipeline/2/artifact/target/devops-sample-0.0.1-SNAPSHOT.jar`
> 
> So, how about creating a new API for this. For instance:
> 
> `GET /kapis/devops.kubesphere.io/v1alpha2/pipelineruns/demo-artifact-tt85n/artifacts?filePath=xxx.log`
> 
> * pros
>   
>   * Easy to understand, easy to memory, shorter
>   * Less to expose the backend tech
>   * Have the same permission control pattern (`pipelineruns/artifacts`)
> * cons
>   
>   * Extra work: both the backend and frontend

Thank you for your suggestion, it is more reasonable to use this method to deal with the problem.

**Comment by LinuxSuRen** [comment]
> Thank you for your suggestion, it is more reasonable to use this method to deal with the problem.

I have created the corresponding issues (both backend and frontend)
* https://github.com/kubesphere/ks-devops/issues/449
* https://github.com/kubesphere/console/issues/2920

I'm going to hold this PR due to contributors might need to take some time to do it. We can unhold it until it is ready.

/hold

**Comment by pixiake** [comment]
> > /cc @LinuxSuRen @JohnNiang
> 
> Thanks for ping us. But a better way is to ping the DevOps SIG team.
> 
> /cc @kubesphere/sig-devops

OK, Got it

**Comment by LinuxSuRen** [comment]
hi @123liubao , I have fixed this issue. Please see https://github.com/kubesphere/ks-installer/pull/1948/files#diff-2af0949adc1f3d7b0f6737b5e6e3d1c870142695e9663faa9ac8641ac5d1fb78R3447

So, I guess we can close this PR.
/close

**Pull request closed without merging**